src/main/java/io/magentys/Agent.java

```diff
@@ -13,7 +13,7 @@
 public class Agent {
 
     private final Memory memory;
-    private List<Any> tools = new ArrayList<>();
+    private final List<Any> tools = new ArrayList<Any>();
```

Comment:
For Java8 we can keep `new ArrayList<Any>`.

Was this an automatic refactoring, or intentional? See my comment above over Java version.